django/db/backends/base/schema.py

```diff
@@ -942,6 +942,9 @@ def _constraint_names(self, model, column_names=None, unique=None,
         Returns all constraint names matching the columns and conditions
         """
         column_names = list(column_names) if column_names else None
+        if column_names and self.connection.features.uppercases_column_names:
+            # Oracle's get_constraints() returns all columns as lower case.
```

Comment:
I suppose one interpretation of this bug is that introspection should do fewer text transformations on the values returned by the databases. It would then be the responsibility of the caller to normalize the name based on `uppercases_column_names` or perhaps use [`normalize_name()`](https://github.com/django/django/blob/master/django/db/backends/oracle/schema.py#L106-L114)

What do you think?